demo/admin/src/App.tsx

```diff
@@ -61,31 +67,29 @@ const categories: AllCategories = [
 
 class App extends React.Component {
     public static render(baseEl: Element): void {
-        Webfontloader.load({
-            google: {
-                families: ["Roboto:100,300,400,500,700"],
-            },
-        });
-
         ReactDOM.render(<App />, baseEl);
     }
 
     public render(): JSX.Element {
         return (
-            <>
-                <SitesConfigProvider
-                    value={{
-                        configs: sitesConfig,
-                        resolveSiteConfigForScope: (configs: Record<string, SiteConfig>, scope: ContentScope) => configs[scope.domain],
-                    }}
-                >
-                    <IntlProvider messages={getMessages()} locale="en">
-                        <LocaleProvider resolveLocaleForScope={(scope: ContentScope) => scope.domain}>
-                            <MuiThemeProvider theme={theme}>
-                                <RouterBrowserRouter>
-                                    <UserProvider>
-                                        <ErrorDialogProvider>
-                                            <ApolloProvider>
+            <ApolloProvider client={apolloClient}>
+                <AuthorizationProvider authorizationManager={authorizationManager}>
+                    <SitesConfigProvider
+                        value={{
+                            configs: sitesConfig,
+                            resolveSiteConfigForScope: (configs: Record<string, SiteConfig>, scope: ContentScope) => configs[scope.domain],
+                        }}
+                    >
+                        <IntlProvider messages={getMessages()} locale="en">
+                            <LocaleProvider resolveLocaleForScope={(scope: ContentScope) => scope.domain}>
+                                <MuiThemeProvider theme={theme}>
+                                    <RouterBrowserRouter>
+                                        <AuthorizationGate
```

Comment:
Why do we need `AuthorizationProvider` and `AuthorizationGate`? Why can't the provider also render the error page?